src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -633,8 +633,8 @@ private void EmitDupExpression(BoundDup expression, bool used)
 
         private void EmitDelegateCreationExpression(BoundDelegateCreationExpression expression, bool used)
         {
-            var mg = expression.Argument as BoundMethodGroup;
-            var receiver = mg != null ? mg.ReceiverOpt : expression.Argument;
+            Debug.Assert(expression.Argument?.Kind != BoundKind.MethodGroup);
+            var receiver = expression.Argument;
```

Comment:
>var receiver = expression.Argument [](start = 12, length = 34)

For the benefit of the next person working with BoundDelegateCreationExpression, please add a detailed comment for the Argument property in the BoundNodes.xml, describing what values could it have at various phases during compilation and how should they be interpreted.